base_import_async/models/base_import_import.py

```diff
@@ -70,42 +72,39 @@ def do(self, fields, options, dryrun=False):
         self._link_attachment_to_job(delayed_job, attachment)
         return []
 
-    @api.model
     def _link_attachment_to_job(self, delayed_job, attachment):
         queue_job = self.env["queue.job"].search(
             [("uuid", "=", delayed_job.uuid)], limit=1
         )
         attachment.write({"res_model": "queue.job", "res_id": queue_job.id})
 
-    @api.model
     @api.returns("ir.attachment")
     def _create_csv_attachment(self, fields, data, options, file_name):
         # write csv
         f = StringIO()
         writer = csv.writer(
             f,
-            delimiter=str(options.get(OPT_SEPARATOR)),
+            delimiter=str(options.get(OPT_SEPARATOR)) or ",",
             quotechar=str(options.get(OPT_QUOTING)),
         )
-        encoding = options.get(OPT_ENCODING, "utf-8")
+        encoding = options.get(OPT_ENCODING) or "utf-8"
         writer.writerow(fields)
         for row in data:
             writer.writerow(row)
         # create attachment
         datas = base64.encodebytes(f.getvalue().encode(encoding))
         attachment = self.env["ir.attachment"].create(
-            {"name": file_name, "datas": datas, "datas_fname": file_name}
+            {"name": file_name, "datas": datas, "store_fname": file_name}
```

Comment:
I'm not sure we should update `store_fname` field which handled internally by the filestore, maybe `name` replaces `datas_fname` now?
```suggestion
            {"name": file_name, "datas": datas}
```